src/shared/components/Button/Button.style.ts

```diff
@@ -103,25 +103,16 @@ const sizeFromProps = ({ size = 'regular', full, hasText }: ButtonStyleProps) =>
 const disabled = ({ disabled }: ButtonStyleProps) =>
   disabled
     ? css`
-        box-shadow: none;
-        color: ${colors.gray[200]};
-        background-color: ${colors.gray[400]};
-        border-color: ${colors.gray[400]};
-        &:hover {
-          color: ${colors.gray[200]};
-          background-color: ${colors.gray[400]};
-          border-color: ${colors.gray[400]};
-        }
-        &:active {
-          color: ${colors.gray[200]};
-          background-color: ${colors.gray[400]};
-          border-color: ${colors.gray[400]};
-        }
+        filter: brightness(50%);
+        pointer-events: none;
       `
     : null
 
 export const StyledIcon = styled(Icon)<IconStyleProps>`
   flex-shrink: 0;
+  width: ${({ size }) => (size === 'large' ? '24px' : '14px')};
```

Comment:
Couple points here:
1. I know it's not specified in Figma yet but designers mentioned to me that the medium button will use a `20px` icon. Forgot to mention that, sorry
2. Since the values here are repeated a couple of times I suggest we define them as variables so we don't need to do too much digging if we decide to change them